gateway/src/main/java/io/zeebe/gateway/EndpointManager.java

```diff
@@ -400,32 +401,36 @@ public static StatusRuntimeException convertThrowable(final Throwable cause) {
 
     if (cause instanceof BrokerErrorException) {
       status = mapBrokerErrorToStatus(((BrokerErrorException) cause).getError());
+      // When there is back pressure, there will be a lot of `RESOURCE_EXHAUSTED` errors and the log
+      // can get flooded. Until we find a way to limit the number of log messages,
+      // let's do not log them.
+      if (status.getCode() != Status.RESOURCE_EXHAUSTED.getCode()) {
+        Loggers.GATEWAY_LOGGER.error(GRPC_ERROR_TEMPLATE + "received error from broker", cause);
+      }
     } else if (cause instanceof BrokerRejectionException) {
       status = mapRejectionToStatus(((BrokerRejectionException) cause).getRejection());
+      Loggers.GATEWAY_LOGGER.trace(GRPC_ERROR_TEMPLATE + "broker rejected request", cause);
     } else if (cause instanceof ClientOutOfMemoryException) {
       status = Status.UNAVAILABLE.augmentDescription(cause.getMessage());
+      Loggers.GATEWAY_LOGGER.error(GRPC_ERROR_TEMPLATE + "gateway out of memory", cause);
     } else if (cause instanceof TimeoutException) { // can be thrown by transport
       status =
           Status.DEADLINE_EXCEEDED.augmentDescription(
               "Time out between gateway and broker: " + cause.getMessage());
+      Loggers.GATEWAY_LOGGER.trace(
+          GRPC_ERROR_TEMPLATE + "request timed out between gateway and broker", cause);
     } else if (cause instanceof GrpcStatusException) {
       status = ((GrpcStatusException) cause).getGrpcStatus();
+      Loggers.GATEWAY_LOGGER.error(GRPC_ERROR_TEMPLATE + "a GrpcStatusException occurred", cause);
```

Comment:
It seems `GrpcStatusException` is an interface which exceptions should implement to determine what status to return. At the moment, some of them are in fact akin to rejections (e.g. we couldn't parse your create workflow payload), so I don't think they should be errors. They're essentially the gateway saying hey this request is wrong before even sending it - but do check, I could be misunderstanding it (even though I wrote it :see_no_evil: Though I think it was a mistake haha)

Reply:
I looked into this. It occurs when trying to deploy a workflow with an unsupported file extension (i.e. not one of: `xml`, `yaml`, `bpmn` or `bpmn20.xml`). It can also occur when a gRPC request cannot be mapped into a BrokerRequest (i.e. a `MsgpackPropertyException` was thrown during this mapping). I wonder whether that second one is a client error or can only occur due to a programming mistake on our side. If they are both client errors, then I can change it to debug level. Otherwise I think the best approach is to replace the wrong file extension to a separate (non gRPCStatusException).

Reply:
Hm, yeah if the broker request cannot be mapped I suppose it could be due to a bug in the gateway itself - I imagine actually it can be both right now, and I'm not sure how trivial it is to differentiate them. Essentially what I'd like is that client errors don't appear in the gateway - so your proposal sounds good, we should differentiate client errors from gateway errors (and log the latter ones as real errors) :+1:

Reply:
As discussed,  the MsgPropertyException can be caused by clients and by system errors. We need to add validation to make sure we handle client errors and system errors independently, but that is outside of the scope of this PR. I will create a new issue for this.

For now, we consider the MsgPackPropertyException a client side exception, because the likelihood of this exception being thrown by a client error is many times greater than it being thrown due to a system error. Therefore, we will reduce its log level in the gateway to debug.